packages/stores/src/account/osmosis/index.ts

```diff
@@ -2040,6 +2044,110 @@ export class OsmosisAccountImpl {
     );
   }
 
+  /**
+   * Method to undelegate from validator set.
+   * @param coin The coin object with denom and amount to undelegate.
+   * @param memo Transaction memo.
+   * @param onFulfill Callback to handle tx fulfillment given raw response.
+   */
+  async sendUndelegateFromValidatorSetMsg(
+    coin: CoinType,
+    memo: string = "",
+    onFulfill?: (tx: any) => void
+  ) {
+    await this.base.signAndBroadcast(
+      this.chainId,
+      "undelegateFromValidatorSet",
+      async () => {
+        const msg = this.msgOpts.undelegateFromValidatorSet.messageComposer({
+          delegator: this.address,
+          coin: {
+            denom: coin.denom.coinMinimalDenom,
+            amount: coin.amount,
+          },
+        });
+
+        return [msg];
+      },
+      memo,
+      {
+        amount: [],
+        gas: this.msgOpts.undelegateFromValidatorSet.gas.toString(),
+      },
+      undefined,
+      (tx) => {
+        if (tx.code == null || tx.code === 0) {
+          // Refresh the balances
+          const queries = this.queriesStore.get(this.chainId);
+          queries.queryBalances
+            .getQueryBech32Address(this.address)
+            .balances.forEach((bal) => {
+              if (
+                bal.currency.coinMinimalDenom === coin.currency.coinMinimalDenom
+              ) {
+                bal.waitFreshResponse();
+              }
+            });
+
+          queries.osmosis?.queryAccountsPositions
+            .get(this.address)
+            .waitFreshResponse();
```

Comment:
Do we need to refresh positions?

We should be intentional when copying and pasting code.